File: Contracts/Model/Product/OrderDto.cs
namespace Contracts.Model.Product;

public sealed class OrderDto
{
    public string OrderId { get; set; }
    public AddressDto Address { get; set; }
    public double Total { get; set; }
    public string? TransactionId { get; set; }
    public string? CurrencyCode { get; set; }
    public string? PayerId { get; set; }
    public double? Shipping { get; set; }
    public DateTimeOffset? TransactionDate { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/UserInteract/orderDto.ts
import {AddressDto} from "@core/interfaces/Product/UserInteract/AddressDto";

export interface OrderDto {
  orderId: string;
  address: AddressDto;
  total: number;
  transactionId: string;
  currencyCode: string;
  payerId: string;
  shipping: number | null;
  shippingDiscount: number | null;
  taxes: number | null;
  transactionDate: string;
}
